src/nvme/tree.c

```diff
@@ -145,6 +145,12 @@ int nvme_read_config(nvme_root_t r, const char *config_file)
 		err = json_read_config(r, config_file);
 		if (!err)
 			r->config_file = strdup(config_file);
+		/*
+		 * The json configuration file is optional,
+		 * so ignore errors when opening the file.
```

Comment:
I am slightly confused by this comment. The only error json_read_config() we will propegate here is when we can't open the file. When the parser fails it will we always be EPROTO. Shouldn't this be the other way round?

Reply:
But it is?
~~~
+               if (err < 0 && errno != EPROTO)
+                       err = 0;
~~~
IE if 'err' == -1 an error occurred, _and_ if the returned errno is not EPROTO we reset 'err' to 0, ie we're blanking out any errors when opening the file.
As the file is optional and opening errors are okay.

Reply:
You are right. Got confused again :)